Win32-network/src/System/Win32/Async/ErrCode.hsc

```diff
@@ -36,3 +39,17 @@ eRROR_PIPE_CONNECTED = #const ERROR_PIPE_CONNECTED
 -- 536
 eRROR_PIPE_LISTENING :: ErrCode
 eRROR_PIPE_LISTENING = #const ERROR_PIPE_LISTENING
+
+type WSAErrCode = Int
+
+wSA_IO_PENDING :: WSAErrCode
+wSA_IO_PENDING = #const WSA_IO_PENDING
+
+foreign import ccall safe "winsock2.hs WSAGetLastError"
+    wsaGetLastError :: IO WSAErrCode
+
+-- Accordin to
+-- <https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-wsagetlasterror>
+-- @FormatMessage@ can be used to get 'WSAErrCode' description.
+wsaFailWith :: String -> WSAErrCode -> IO a
+wsaFailWith fn_name err_code = failWith fn_name (unsafeCoerce err_code)
```

Comment:
`unsafeCoerce` is not an appropriate conversion function. Don't be in the habit of reaching for it, even for FFI. You probably only need `fromIntegral`.

Reply:
right, that's quite bad.